core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/StoreCallbacksImpl.scala

```diff
@@ -125,6 +128,14 @@ object StoreCallbacksImpl extends StoreCallbacks with Logging with Serializable
     schemas.add(ColumnFormatRelation.INTERNAL_SCHEMA_NAME)
     schemas
   }
+
+  override def getHashCodeSnappy(dvd: scala.Any): Int = {
+    partioner.hashValue(dvd)
```

Comment:
See comment above. Definitely need method at least for string to calculate hash without deserialisation. Right now it is deserializing to create string then serializing again in the method body.